MohamedH1998/99NamesAPI
# Refactor Names API service to follow proper API standards

## What has changed

- Updated API routes to follow proper versioning standards 
  - If you go through the names controller you will notice that I've made some changes to the routing. Specifically I've added API version via the URI. This is important if we want to expand our API in the future, we can simple do `/api/v2/names/...`
- Refactored HTTP response codes to follow proper API standards.
  - I've noticed that we were using 400 response codes for instances where our API fails to find a name. Those should actually be 404s
- Removed the routes that gets a random name
  - This was causing a bug where if I did something like `/api/v1/names/something/bad/here/1234/yes/` it would still return a response and this can be very bad!!!
- There were no ESLINT rules set so I went ahead and created one so we all follow the same coding guidelines. It makes our lives easier.. don't worry too much about your editor changing etc these are local to this project and wont affect any other projects

## How to test changes

Testing the code should be simple. There are no unit tests yet so for now you will need to do the following:
1. Pull this PR, and confirm that the `/` and `*` no longer work.
2. confirm that the names and name routes return a valid response
3. confirm that putting an invalid route returns an error
4. confirm that putting an incorrect range in the range api returns a 400 error stating that `Names do not seem to exist within the given range of ids.`
